File: Power Query/300-399/324/PQ 324.py
import pandas as pd
import numpy as np

path = "300-399/324/PQ_Challenge_324.xlsx"

input = pd.read_excel(path, usecols="A:B", nrows=22)
test = pd.read_excel(path, usecols="D:G", nrows=8).fillna(0)
test['South Avenue'] = test['South Avenue'].astype(int)

input['Store'] = input.loc[input['Data1'] == 'Store', 'Data2']
input['Store'] = input['Store'].ffill()
input['Visit Date'] = input.loc[input['Data1'] == 'Visit Date', 'Data2']
input['Visit Date'] = input['Visit Date'][::-1].ffill()[::-1]
mask = (input['Data2'] != input['Store']) & (input['Data2'] != input['Visit Date'])
df = input.loc[mask, ['Data2', 'Store']]
df = df.assign(Name=df['Data2'].str.split(', ')).explode('Name').drop(columns='Data2')
result = df.groupby(['Name', 'Store']).size().unstack(fill_value=0)
result.loc['Total'] = result.sum()
result['Total'] = result.sum(axis=1)
result = result.reset_index()
result.columns.name = None

print(result.equals(test))

File: Power Query/300-399/324/PQ 324.R
library(tidyverse)
library(readxl)

path = "Power Query/300-399/324/PQ_Challenge_324.xlsx"
input = read_excel(path, range = "A1:B22")
test  = read_excel(path, range = "D1:G9") %>%
  mutate(across(everything(), ~replace_na(.x, 0)))

result = input %>%
  mutate(Store = ifelse(Data1 == "Store", Data2, NA_character_)) %>%
  fill(Store) %>%
  mutate(`Visit Date` = ifelse(Data1 == "Visit Date", Data2, NA_character_)) %>%
  fill(`Visit Date`, .direction = "up") %>%
  filter(Data2 != Store, Data2 != `Visit Date`) %>%
  select(-c(Data1, `Visit Date`)) %>%
  separate_longer_delim(Data2, ", ") %>%
  rename(Name = Data2) %>%
  count(Name, Store) %>%
  pivot_wider(names_from = Store, values_from = n, values_fill = 0) %>%
  janitor::adorn_totals(c("row", "col"))

all.equal(result, test, check.attributes = FALSE)
# TRUE